### ISSUE TITLE
The --no-bundler flag throws errors

### ISSUE THREAD

**Issue by robbiewxyz** [expo/expo#30569]
### Summary

This command line flag doesn't have any tests and has probably hasn't seen any love recently and has broken as other changes have been made to the build process.

For e.g. Android the command `expo run:android --no-bundler` fails with the following:

```
Waiting on http://localhost:8081
› Opening exp+pier26://expo-development-client/?url=http%3A%2F%2F192.168.0.112%3A8081 on Pixel_5_API_34
Failed to stop dev server (bundler: metro)
CommandError: Timeout waiting for 'metro' dev server to close
```

### What platform(s) does this occur on?

Android, iOS

### SDK Version

51.0.21

### Environment

  expo-env-info 1.2.0 environment info:
    System:
      OS: macOS 14.4.1
      Shell: 5.9 - /opt/homebrew/bin/zsh
    Binaries:
      Node: 20.12.2 - ~/Library/Caches/fnm_multishells/17673_1721630352859/bin/node
      npm: 10.5.0 - ~/Library/Caches/fnm_multishells/17673_1721630352859/bin/npm
      Watchman: 2024.04.22.00 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.15.2 - /opt/homebrew/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 23.4, iOS 17.4, macOS 14.4, tvOS 17.4, visionOS 1.1, watchOS 10.4
      Android SDK:
        API Levels: 34, 35
        Build Tools: 33.0.1, 34.0.0, 35.0.0
        System Images: android-34 | Google APIs ARM 64 v8a, android-35 | Google Play Experimental 16k Page Size ARM 64 v8a, android-35 | Google APIs Experimental 16k Page Size ARM 64 v8a
    IDEs:
      Android Studio: 2023.2 AI-232.10300.40.2321.11668458
      Xcode: 15.3/15E204a - /usr/bin/xcodebuild
    npmPackages:
      react-dom: 18.2.0 => 18.2.0 
    Expo Workflow: bare

### Minimal reproducible example

The behavior of this command line flag is almost certainly not dependent on project setup. Core devs, do you mind trying this flag one one of your clean environments? If it doesn't reproduce right away I'll take the time to create a repository.

**Comment by albertorg** [comment]
Same error

**Comment by byCedric** [comment]
This has been fixed by #30432. I will backport it to SDK 51 cc @brentvatne
